### PR TITLE
[REVIEW] Add a bunch of missing typecasting to/from strings as well as tests

### LINKED ISSUE #1714: [BUG] Runtime error when converting a String column to datetime64 type
**Describe the bug**
When trying to convert a string series to a datetime64, the following error is displayed:

image

**Steps/Code to reproduce bug**
Run the code above in RAPIDS v0.7

**Expected behavior**
The column should be converted to datetime64 without issues.

**Environment details (please complete the following information):**
 - Environment location: [Bare-metal, Docker, Cloud(specify cloud provider)]
 - Method of cuDF install: [conda, Docker, or from source]
   - If method of install is [Docker], provide `docker pull` & `docker run` commands used
 - Please run and attach the output of the `cudf/print_env.sh` script to gather relevant environment details
 

**Additional context**

### LINKED ISSUE #1901: [FEA] Uncomment and fix string related typecasting
The relevant methods have been made in nvstrings to finish the typecasts to/from strings so we should uncomment the code and write relevant tests.

### PR DESCRIPTION
**Pull request by kkraus14**
Fixes #1714
Fixes #1901

### CODE REVIEW

File: ci/gpu/build.sh [hunk 1]
```diff
@@ -18,9 +18,9 @@ export CUDA_REL=${CUDA_VERSION%.*}
 # Set home to the job's workspace
 export HOME=$WORKSPACE
 
-# Set versions of packages needed to be grabbed
-export NVSTRINGS_VERSION=0.7.*
-export RMM_VERSION=0.7.*
+# Parse git describe
+export GIT_DESCRIBE_TAG=`git describe --tags`
+export MINOR_VERSION=`echo $GIT_DESCRIBE_TAG | grep -o -E '([0-9]\.[0-9])'`
```
Comment (raydouglass) [inline ci/gpu/build.sh]: You should probably do `([0-9]+\.[0-9]+)` to capture `10.0` or '0.11`

File: cpp/tests/io/csv/csv_test.cu [hunk 2]
```diff
@@ -848,10 +848,10 @@ TEST(gdf_csv_test, Writer)
     std::string csv((std::istreambuf_iterator<char>(infile)), std::istreambuf_iterator<char>());
     std::string verify =
         "\"boolean\",\"integer\",\"float\",\"string\"\n"
-        "true,1,1,\"one\"\n"
+        "true,1,1.0,\"one\"\n"
         "false,2,2.25,\"two\"\n"
         "false,3,3.5,\"three\"\n"
         "true,4,4.75,\"four\"\n"
-        "false,5,5,\"five\"\n";
+        "false,5,5.0,\"five\"\n";
```
Comment (shwina) [inline cpp/tests/io/csv/csv_test.cu]: Just curious why these changes were needed

File: ci/gpu/build.sh [hunk 3]
```diff
@@ -34,7 +34,7 @@ nvidia-smi
 
 logger "Activate conda env..."
 source activate gdf
-conda install -c rapidsai/label/cuda${CUDA_REL} -c rapidsai-nightly/label/cuda${CUDA_REL} rmm=${RMM_VERSION} nvstrings=${NVSTRINGS_VERSION}
+conda install "rmm=$MINOR_VERSION.*" "nvstrings=$MINOR_VERSION.*"
```
Comment (raydouglass) [inline ci/gpu/build.sh]: I think you need `cudatoolkit=${CUDA_REL}` installed here too.

File: cpp/tests/io/csv/csv_test.cu (hunk 2 above)
Comment (kkraus14) [reply cpp/tests/io/csv/csv_test.cu]: There was a change in the float to string code in nvstrings 0.8 that made "5" --> "5.0"

**Comment by kkraus14** [comment]
@mjsamoht could you review/approve the changes from @j-ieong?

**Comment by mike-wendt** [comment]
rerun tests

**Pull request merged**